core-data/selectors.js

```diff
@@ -145,3 +145,28 @@ export const getEntityRecords = createSelector(
 export function getThemeSupports( state ) {
 	return state.themeSupports;
 }
+
+/**
+ * Returns the embed preview for the given URL.
+ *
+ * @param {Object} state    Data state.
+ * @param {string} url      Embedded URL.
+ *
+ * @return {Mixed} Undefined if the preview has not been fetched, false if the URL cannot be embedded, array of embed preview data if the preview has been fetched.
+ */
+export function getPreview( state, url ) {
+	const preview = state.embedPreviews[ url ];
+
+	if ( ! preview ) {
+		return preview;
+	}
+
+	const oEmbedLinkCheck = '<a href="' + url + '">' + url + '</a>';
+
+	if ( oEmbedLinkCheck === preview.html ) {
```

Comment:
I wish there was a more reliable way to determine this from the server? Like a response property or header. Constructing a string URL to compare seems rather fragile.